linux/devdoc/threadpool_linux_requirements.md

```diff
@@ -0,0 +1,350 @@
+
+```mermaid
+graph TD
+    A(insert_idx == consume_idx) --> |worker thread runs first| B(consume_idx increment)
+    A--> |main thread runs first| D(no gaps) -->F
+    B--> C(gaps occured)-->E(memmove)-->F(resize done)
+    
+```
+
+## Exposed API
+```C
+
+typedef struct THREADPOOL_TAG* THREADPOOL_HANDLE;
+typedef struct TIMER_INSTANCE_TAG* TIMER_INSTANCE_HANDLE;
+
+#define THREADPOOL_OPEN_RESULT_VALUES \
+    THREADPOOL_OPEN_OK, \
+    THREADPOOL_OPEN_ERROR
+
+MU_DEFINE_ENUM(THREADPOOL_OPEN_RESULT, THREADPOOL_OPEN_RESULT_VALUES)
+
+typedef void (*ON_THREADPOOL_OPEN_COMPLETE)(void* context, THREADPOOL_OPEN_RESULT open_result);
+typedef void (*THREADPOOL_WORK_FUNCTION)(void* context);
+
+MOCKABLE_FUNCTION(, THREADPOOL_HANDLE, threadpool_create, EXECUTION_ENGINE_HANDLE, execution_engine);
+MOCKABLE_FUNCTION(, void, threadpool_destroy, THREADPOOL_HANDLE, threadpool);
+
+MOCKABLE_FUNCTION(, int, threadpool_open_async, THREADPOOL_HANDLE, threadpool, ON_THREADPOOL_OPEN_COMPLETE, on_open_complete, void*, on_open_complete_context);
+MOCKABLE_FUNCTION(, void, threadpool_close, THREADPOOL_HANDLE, threadpool);
+
+MOCKABLE_FUNCTION(, int, threadpool_schedule_work, THREADPOOL_HANDLE, threadpool, THREADPOOL_WORK_FUNCTION, work_function, void*, work_function_context);
+
+MOCKABLE_FUNCTION(, int, threadpool_timer_start, THREADPOOL_HANDLE, threadpool, uint32_t, start_delay_ms, uint32_t, timer_period_ms, THREADPOOL_WORK_FUNCTION, work_function, void*, work_function_context, TIMER_INSTANCE_HANDLE*, timer_handle);
+
+MOCKABLE_FUNCTION(, int, threadpool_timer_restart, TIMER_INSTANCE_HANDLE, timer, uint32_t, start_delay_ms, uint32_t, timer_period_ms);
+
+MOCKABLE_FUNCTION(, void, threadpool_timer_cancel, TIMER_INSTANCE_HANDLE, timer);
+
+MOCKABLE_FUNCTION(, void, threadpool_timer_destroy, TIMER_INSTANCE_HANDLE, timer);
+
+```
+###  threadpool_create
+
+```C
+MOCKABLE_FUNCTION(, THREADPOOL_HANDLE, threadpool_create, EXECUTION_ENGINE_HANDLE, execution_engine);
+```
+
+`threadpool_create` creates a new threadpool.
+
+`threadpool_create` shall allocate memory for a threadpool object and on success return a non-`NULL` handle to it. 
+
+If `execution_engine` is `NULL`, `threadpool_create` shall fail and return `NULL`. 
+
+`threadpool_create` shall create a `SM_HANDLE` by calling `sm_create`.
+
+`threadpool_create` shall initialize thread count parameters by calling `execution_engine_linux_get_parameters` with parameter `execution_engine`. 
```

Comment:
I think this wants to say "shall get the min/max thread parameters from the execution engine"

Reply:
yes, modified